diff --git a/node_modules/laji-map/lib/utils.js b/node_modules/laji-map/lib/utils.js
index a01b7ed..aedc06e 100644
--- a/node_modules/laji-map/lib/utils.js
+++ b/node_modules/laji-map/lib/utils.js
@@ -38,7 +38,7 @@ function convertLatLng(latlng, from, to) {
         switch (format) {
             case "EPSG:2393": return globals_1.EPSG2393String;
             case "EPSG:3067": return globals_1.EPSG3067String;
-            default: return proj4_1.default.defs(format);
+            default: return proj4_1.defs(format);
         }
     }
     var validator = undefined;
@@ -50,7 +50,7 @@ function convertLatLng(latlng, from, to) {
     }
     if (validator && validator.formatter)
         latlng = latlng.map(function (c) { return "" + c; }).map(function (c, i) { return +validator[i].formatter(c); });
-    var converted = proj4_1.default(formatToProj4Format(from), formatToProj4Format(to), reverseCoordinate(latlng));
+    var converted = proj4_1(formatToProj4Format(from), formatToProj4Format(to), reverseCoordinate(latlng));
     return (to === "WGS84") ? converted : converted.map(function (c) { return parseInt(c); });
 }
 exports.convertLatLng = convertLatLng;
