diff --git a/node_modules/ngx-webstorage/fesm5/ngx-webstorage.js b/node_modules/ngx-webstorage/fesm5/ngx-webstorage.js
index a713fc6..a8fe944 100644
--- a/node_modules/ngx-webstorage/fesm5/ngx-webstorage.js
+++ b/node_modules/ngx-webstorage/fesm5/ngx-webstorage.js
@@ -151,12 +151,20 @@ var StrategyCacheService = /** @class */ (function () {
 
 var LOCAL_STORAGE = new InjectionToken('window_local_storage');
 function getLocalStorage() {
+  try {
     return (typeof window !== 'undefined') ? window.localStorage : null;
+  } catch (e) {
+    return null;
+  }
 }
 var LocalStorageProvider = { provide: LOCAL_STORAGE, useFactory: getLocalStorage };
 var SESSION_STORAGE = new InjectionToken('window_session_storage');
 function getSessionStorage() {
+  try {
     return (typeof window !== 'undefined') ? window.sessionStorage : null;
+  } catch (e) {
+    return null;
+  }
 }
 var SessionStorageProvider = { provide: SESSION_STORAGE, useFactory: getSessionStorage };
 
